shop/forms/checkout.py

```diff
@@ -168,9 +168,12 @@ def form_factory(cls, request, data, cart):
                 # prevent adding the same address twice
                 all_field_names = [f.name for f in cls.get_model()._meta.get_fields()]
                 filter_args = dict((attr, val) for attr, val in address_form.data.items()
-                                   if attr in all_field_names and val)
+                                   if attr in all_field_names)
```

Comment:
Removing the ``and val`` introduced a regression.
Please go to the demo and enter an a new shipping address.
Since the address form is uploaded twice (the first time to determine context for shipping costs, and the second time to proceed), without the ``and val`` we would add the address twice.

Therefore I'll re-add it.